### PR TITLE
Rewrite/update styling of local search

### PR DESCRIPTION
**Pull request by thaJeztah**
See individual commits for details

### Homepage search before this change:

Screenshot 2020-10-11 at 14 13 03
Screenshot 2020-10-11 at 14 12 50

### Homepage search after this change:

Screenshot 2020-10-11 at 14 11 46
Screenshot 2020-10-11 at 14 11 31

### CODE REVIEW

**Comment by thaJeztah** [comment]
Functionality / style changes for search:

- Rewrite javascript to remove some buggy / hard to follow parts
- Reduced styles used in HTML markup; many classes were not needed
- Make some output conditional, e.g., if the "description" and "title" are the same, don't print the description
- Don't output the URL in the autocomplete output
- Fix "active" item style in dark mode
- Remove "Description" label in the output; it should be clear from the output that it's a description
- Replaced "Keywords" label in output with a "tag" icon
- Reduced font size and spacing (can be tweaked if needed)
- Reduced size, and rephrased  "summary" (number of results shown) output

**Comment by thaJeztah** [comment]
@anusha-ragunathan @StefanScherer ptal

File: _scss/_content.scss [hunk 1]
```diff
@@ -13,7 +13,9 @@
  */
 
 pre {
-    background-color: #F5F8FA!important;
+    background-color: #f5f5f5!important;
```
Comment (StefanScherer) [inline _scss/_content.scss:16]: minor not: Should it be `#f5f8fa` ?

File: _scss/_night-mode.scss [hunk 2]
```diff
@@ -105,105 +75,55 @@ body.night {
...
-    .autocompleteTitle {
-        font-weight: bold;
-    }
-    .autocompleteUrl,
-    .autocompleteKeywords,
-    .autocompleteDescription {
-        line-height: normal;
+    .autocompleteResult {
+        &.selected {
+            background-color: #0f161e;
```
Comment (StefanScherer) [inline _scss/_night-mode.scss:112]: This looks a bit dark for the selected search result:

navigating-search-result-dark


Maybe a bit lighter. 🤔 

```suggestion
            background-color: #1f262e;
```

File: _scss/_utilities.scss [hunk 3]
```diff
@@ -125,227 +113,90 @@ img.inline {
...
-.autoCompleteResult {
+.autocompleteResult {
     border-bottom: 1px solid rgba(203, 205, 209, 0.4);
+    cursor: pointer;
     padding: 15px;
-
-    &:hover {
-        cursor: pointer;
+    &.selected {
+      background-color: #f5f5f5;
```
Comment (StefanScherer) [inline _scss/_utilities.scss:155]:
```suggestion
      background-color: #f5f8fa;
```

**Review by StefanScherer — CHANGES_REQUESTED** [review]
Looks very good so far. I only have some suggestions to change the background color of the selected search result to make it a bit more visible what is selected.

File: _scss/_content.scss (hunk 1 above)
Comment (thaJeztah) [reply _scss/_content.scss:16]: oh! let me check what color was previously active. There were some rules that were duplicate, so it may be that I took the one that had the highest priority (and thus was previously active).

File: _scss/_night-mode.scss (hunk 2 above)
Comment (thaJeztah) [reply _scss/_night-mode.scss:112]: Yes, I also found it too dark; this was the color that was "meant" to be picked so I kept that color, but let me tweak it a bit to my own taste 😅

File: _scss/_utilities.scss (hunk 3 above)
Comment (thaJeztah) [reply _scss/_utilities.scss:155]: Updated with your suggestion

**Comment by usha-mandya** [comment]
The search results still say result instead of results. 

image

**Comment by thaJeztah** [comment]
oh! forgot to update that; fixed

**Review by usha-mandya — APPROVED** [review]
LGTM. Thanks @thaJeztah

**Comment by usha-mandya** [comment]
Giving a chance for @StefanScherer to take a quick look before I merge.

**Review by StefanScherer — APPROVED** [review]
LGTM 🎉
